public/js/roles/roles-list.controller.js

```diff
@@ -64,5 +65,30 @@
                 vm.getRoles(vm.page);
             });
         }
+
+        function deleteRole (roleId) {
+            api.getUsers().then(function (res) {
+                const usersHasRole = res.data.filter(e => e.roles.includes(roleId));
+                const targetRole = vm.items.filter(e => { return e._id === roleId; })[0];
+
+                const modal = $uibModal.open({
+                    component: 'appDeleteModal',
+                    resolve: {
+                        title: () => gettextCatalog.getString('Remove role {{name}} ? - {{usersCount}} user(s) affected', { name: targetRole.name, usersCount: usersHasRole.length }),
+                        delete: () => function () {
+                            if (usersHasRole.length > 0) {
+                                usersHasRole.forEach(u => api.deleteUserRole(u._id, roleId));
```

Comment:
This should be done on the server side during the request that deletes the role.
1) It will be much faster than doing one API request per user.
2) It's the server's responsibility to ensure data integrity